### ISSUE TITLE
Upcoming Release `5.0.0` Pending

### ISSUE THREAD

**Issue by Marak** [Marak/faker.js#692]
### If anyone has any objections / concerns / questions please respond soon so we can address these issues before release. We anticipate not having any major API changes or breaking issues.

_______
**New API Methods**

Added faker.date.soon https://github.com/Marak/faker.js/pull/487

Added aba routing number https://github.com/Marak/faker.js/pull/423

Added method `faker.random.float`  https://github.com/Marak/faker.js/pull/530

Added `addess.nearbyGPSCoordinate` method https://github.com/Marak/faker.js/pull/378

Added `commerce.creditCardNumber` https://github.com/Marak/faker.js/pull/382

Added `git` faker module https://github.com/Marak/faker.js/pull/662

**Locality Updates**

Added South African `za` Locale https://github.com/Marak/faker.js/pull/517

Added `nl_BE` locale https://github.com/Marak/faker.js/pull/529/

Added `fa` locality https://github.com/Marak/faker.js/pull/391

Added `ro` locality https://github.com/Marak/faker.js/pull/327

Added `ar` locality ( partial ) https://github.com/Marak/faker.js/pull/505

Updated to to 500 US Census names for `en` locality https://github.com/Marak/faker.js/pull/360

Added a few female names ( `sv` locale ) https://github.com/Marak/faker.js/pull/397

Wrong cell phone format for `pl` locale https://github.com/Marak/faker.js/pull/655

Added `address.zipCodeByState` method https://github.com/Marak/faker.js/pull/600

Added `ru` locale hacker directory https://github.com/Marak/faker.js/pull/476

**API Changes**

`random.alphaNumeric` now accepts count as a argument https://github.com/Marak/faker.js/pull/363

**API Fixes**

`internet.userAgent` missing random seed https://github.com/Marak/faker.js/issues/521

`commerce.price` can no longer default to 0 https://github.com/Marak/faker.js/pull/446

Generation of float numbers  https://github.com/Marak/faker.js/issues/401 https://github.com/Marak/faker.js/issues/478

Added an explict `faker.setLocale` method to resolve ES6 import issue https://github.com/Marak/faker.js/pull/488

`this` missing in password generator https://github.com/Marak/faker.js/pull/386  

**Distribution**

Update bundled jquery for demo https://github.com/Marak/faker.js/issues/489

**Comment by jonny64** [comment]
How to install it via npm beta package?

**Comment by jonny64** [comment]
We use faker.js to generate users in locale ru
This generates invalid user with fistName in en and lastName in ru locale
This user cannot be used further due to validation checks

**Comment by j0k3r** [comment]
Just to let you know we tested Faker using the master branch on all our repos and it works great :+1:

**Comment by kuncevic** [comment]
It would be easier to test if you would publish like `RC` version or something tagging that say `"next"`  so we can easily pull that from npm like `npm i faker@next`

**Comment by j0k3r** [comment]
Or maybe just a pre-release on GitHub like `5.0.0-rc1` which will be pushed to NPM too.

**Comment by jonny64** [comment]
Hi, I've added test for my first comment
for wrong locale internet.userName https://github.com/Marak/faker.js/pull/715
and fixed https://github.com/Marak/faker.js/pull/713

**Comment by kuncevic** [comment]
Any plans of resolving this one https://github.com/Marak/faker.js/issues/643 
I am having such a case where I have to use `faker.fake()` passing string that contained params:
```
template = "{{date.between("2018-01-01", "2018-12-31")}}";
faker.fake(template) // does not work
```
ref: https://github.com/Marak/faker.js/blob/be8cd96541ba277941f4489fba27fc96296ed15c/lib/date.js#L64

However this works 
```
template = '{{random.number({ "min": 50, "max": 100, "precision": 1 })}}'
faker.fake(template) 
```
ref: https://github.com/Marak/faker.js/blob/8f76fa2ca1de4e20688d289302656f4de0c9ad11/lib/random.js#L23
@Marak  Do you think it is worthwhile to turn input params in to Options similar like `random.number` does?

**Comment by Marak** [comment]
@kuncevic 

 I am happy to try and resolve whatever problem you may be facing, but I don't fully understand your comment. "`does not work`" and "`works`" are hard for me to comprehend.

If anyone can open a PR for #643 which includes the failing unit tests I'll be glad to get it merged in. Please feel free to add more comments or respond directly in that issue.

If you feel a method signature in faker should be changed or updated please open a new issue and we can discuss it. 

We should not be updating method signatures only in order to make `faker.fake` work...we should be resolving the root cause.

**Comment by kuncevic** [comment]
The problem with `faker.fake` that it does not handle the all possible variations of parameters that can be passed to functions due to different signatures on each functions, some accepted Options (e.g `[random.number](https://github.com/Marak/faker.js/blob/8f76fa2ca1de4e20688d289302656f4de0c9ad11/lib/random.js#L23)`) some not e.g ([`date.between](https://github.com/Marak/faker.js/blob/be8cd96541ba277941f4489fba27fc96296ed15c/lib/date.js#L64)`) that does not handling use cases like `faker.fake('{{date.between("2018-01-01", "2018-12-31")}}')`

**Comment by kuncevic** [comment]
@Marak  What was the problem with comment? :/

**Comment by danjrichardson** [comment]
`faker.random.arrayElements` was also pulled in from pull #366 over a year ago.

**Comment by damienwebdev** [comment]
I believe faker.unique is also in there as well.

**Comment by lucianmldvn** [comment]
Hi,

Any idea when will version 5 be released? I am using faker.js in a Adobe XD Plugin i'm making and I could use the additional features and data.

**Comment by thundo** [comment]
Same for `random.hexaDecimal` added on Jun 2017. Any progress for a (pre-)release?

Thanks

**Comment by JesseObrien** [comment]
Would love a pre-release for the new `address.nearbyGPSCoordinate`. Any updates?

**Comment by jontroncoso** [comment]
same for `address.zipCodeByState`. 

### PLEASE RELEASE!!

_plz_

Can I put a dollar in a box somewhere?

**Comment by liquidvisual** [comment]
Is this still happening? 6 months is a long time

**Comment by pbassut** [comment]
Where did the author go?
It's been a while. I'm waiting on the `faker.number.float()` and all the other great stuff.
Would be great if we could get a RC at least.

**Comment by Marak** [comment]
I'd really like to be putting more hours in but it's a lot of work with no pay.

I'm hoping I can put in more hours this month and get a release out.

If we can get few more sponsors on https://opencollective.com/fakerjs I can devote more time to the project.

**Comment by pbassut** [comment]
Just became a backer and I encourage everyone to do so. It's a like a cup of coffee per month for a great library. Come on you guys and gals

**Comment by AbdealiJK** [comment]
@Marak do let us know if we can help in making this release possible.
Considering there has been no objection in this issue - as there any blockers for an RC release ?

If not, can we help (in terms of developing effort or so) in any way for a release to be put into npm ?

**Comment by andreialecu** [comment]
For anyone waiting on this, you can also add the github master branch as your dependency:

package.json
```
    "faker": "Marak/faker.js",
```

**Comment by WillGibson** [comment]
I hit the https://github.com/Marak/faker.js/issues/639 issue after trying to use it based on the documentation.

Following @pbassut's lead I am also now a backer. Totally worth it, I use faker.js loads in my tests.

@Marak, time is also a finite resource in my world, but do reach out with things I can do to help move the next release along.

**Comment by jonrimmer** [comment]
I would be happy to contribute, but the funding model is a little unclear. There is already a submitted and paid expense of $600 for the v5.0.0 release from last year: https://opencollective.com/fakerjs/expenses/3972. And yet the 5.0.0 release has not happened, so what exactly did this money pay for?

**Comment by Marak** [comment]
@jonrimmer I've deleted your comment and removed your ability to contribute to the project.

I am not responsible for managing your contribution expectations or the conditions to which you may or may not contribute.
